### DISCUSSION TITLE
Can you use next script to render html

### DISCUSSION THREAD

**Discussion by mbuguanewton** [vercel/next.js#40353] [Help] [answered]
I have a script to embed through my CMS. 

`<script>window['stg'] = window['stg'] || {};window['stg'].optedOutCheckboxText="You’re opted out. Uncheck to opt in.";window['stg'].optedInCheckboxText="Opt out of data collection and usage";window['stg'].isGlobal=false;window['stg'].instanceHostname="https://<domain>.containers.piwik.pro";</script>

<div id="_stg_opt_out_iframe_content" style="display: none;"><p>Privacy on this site: We collect and process your data on this site to better understand how it is used. You can opt-in or opt-out at any time.</p><input id="_stg_optout_checkbox" onclick="window.parent.storeUserOptOutPreferences();" type="checkbox"><label id="_stg_optout_checkbox_label" for="_stg_optout_checkbox"></label></div>

<script src="https://<domain.com>/privacy/script.v2.js"> </script>`

Is it possible to render the above script using next/script and include the HTML or do I need to separate the scripts and the HTML? How can you render this on a Next JS page?

**Comment by PsyGik** [comment, marked as answer]
You can use the `Script` from `'next/script'`. Something along the lines of (untested code):

```js
import React from 'react';
import Script from 'next/script';
const PrivacyNotice = () => (
  <>
    <Script
      id="my-script-ir"
      strategy="afterInteractive"
      dangerouslySetInnerHTML={{
        __html: `
    (function(w){
    // do things with DOM
    w['stg'] = w['stg'] || {};
    w['stg'].optedOutCheckboxText="You’re opted out. Uncheck to opt in.";
    w['stg'].optedInCheckboxText="Opt out of data collection and usage";
    w['stg'].isGlobal=false;
    w['stg'].instanceHostname="https://.containers.piwik.pro";
    })(window);
  `,
      }}
    />
    Privacy on this site: We collect and process your data on this site to
    better understand how it is used. You can opt-in or opt-out at any time.
    <Script src="https:///privacy/script.v2.js" strategy="lazyOnload" />
  </>
);
```

and use the `PrivacyNotice` component anywhere in your app.

Choose the [strategy](https://nextjs.org/docs/basic-features/script#strategy) that works best for your use-case.

**Reply by icyJoseph** [reply]
This ought to work. Just remember to verify the behaviour is as you want it, when doing, SPA navigations and when the script un-mounts.

**Reply by mbuguanewton** [reply]
It makes sense. Just thought the script always needs to be inside the Head tag `next/head`. Thanks, @PsyGik and @icyJoseph

**Reply by icyJoseph** [reply]
Not at all, https://nextjs.org/docs/basic-features/script:

> The Next.js Script component, [next/script](https://nextjs.org/docs/api-reference/next/script), is an extension of the HTML <script> element. It enables developers to set the loading priority of third-party scripts anywhere in their application, **outside next/head**, saving developer time while improving loading performance.
